bundles/org.openhab.binding.bluetooth/src/main/java/org/openhab/binding/bluetooth/discovery/internal/BluetoothDiscoveryProcess.java

```diff
@@ -122,6 +127,15 @@ public DiscoveryResult get() {
         return result;
     }
 
+    private boolean isAddressAvailable(BluetoothAddress address) {
+        // if a device with this address has a handler on any of the adapters, we abandon discovery
+        return adapters.stream().filter(adapter -> adapter.hasDevice(address))
+                // get adapter's corresponding device
+                .map(adapter -> adapter.getDevice(address))
+                // make sure nothing is listening to any of them
+                .noneMatch(d -> d.hasListeners());
```

Comment:
for the next iteration :-)

```suggestion
                .noneMatch(BluetoothDevice::hasListeners);
```